tests/integration/high_availability/application_charm/src/continuous_writes.py

```diff
@@ -37,13 +37,14 @@ def continuous_writes(database_config: Dict, table_name: str, starting_number: i
                     f"INSERT INTO `{table_name}`(number) VALUES ({next_value_to_insert})"
                 )
         except mysql.connector.errors.DatabaseError as e:
-            # errno 2003: can't connect to mysql database
-            if e.errno != 2003:
-                raise
+            if e.errno == 1062:
+                with MySQLConnector(database_config) as cursor:
+                    cursor.execute(f"SELECT max(number) FROM `{table_name}`")
+                    result = cursor.fetchall()
+                    next_value_to_insert = result[0][0] + 1
```

Comment:
LGTM, could be simplified the next time by using column with `AUTO_INCREMENT` and simply insert anything and checking primary key value (always in innodb_buffer_pool).